Pandas Part 1 — Exercises

Section 1 — Insurance Dataset

In [2]:
import pandas as pd

In [3]:
import numpy as np

1) Import the data and check the first few rows of the dataset. Also check the last few rows of the dataset.


In [5]:
df = pd.read_csv("../data/insurance.csv")
df.tail()

age     sex    bmi  children smoker     region     charges
1333   50    male  30.97         3     no  northwest  10600.5483
1334   18  female  31.92         0     no  northeast   2205.9808
1335   18  female  36.85         0     no  southeast   1629.8335
1336   21  female  25.80         0     no  southwest   2007.9450
1337   61  female  29.07         0    yes  northwest  29141.3603

2) Check if there are any missing values ​​in the dataset.

In [6]:
df.isnull().values.any()

np.False_

3) How much memory is the dataset consuming?

In [7]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


4) Generate descriptive statistics for the numerical columns of the dataset. What is the median of the BMI column? And the mean? Which of these values ​​is larger? What does this mean?

In [ ]:
df.describe()  



age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

## Q4 - mean=30.663397  
median = 30.400000 

Mean BMI is slightly greater than the median BMI which means there are some higher BMI values pulling the average up

5) What is the proportion of smokers in our dataset? And of non-smokers?

In [12]:
df["smoker"].value_counts(normalize=True)*100


smoker
no     79.521674
yes    20.478326
Name: proportion, dtype: float64

6) Is there any difference in the charges for smokers and non-smokers? If so, what do you think is the reason for this?

In [ ]:
df.groupby("smoker")["charges"].mean()




smoker
no      8434.268298
yes    32050.231832
Name: charges, dtype: float64

## Q6 - Smokers have substantially higher average charges, likely because smoking increases health risks and expected medical costs, which raises insurance pricing.

7) What is the average BMI for smokers? Is it very different from that observed for non-smokers? If so, why do you think this is the case?

In [16]:
df.groupby("smoker")["bmi"].mean()

smoker
no     30.651795
yes    30.708449
Name: bmi, dtype: float64

## Q7 - BMI is very similar between smokers and non-smokers

8) Is there any difference in charges for those who have children? Identify if having children seems to result in higher charges by looking at the average charges for those with 0 children and the average for those with 1 or more children.

In [17]:
df.groupby("children")["charges"].mean()

children
0    12365.975602
1    12731.171832
2    15073.563734
3    15355.318367
4    13850.656311
5     8786.035247
Name: charges, dtype: float64

## Q8 - Individuals with at least one child show higher average charges than those without children, possibly because families tend to use more healthcare services, increasing total insurance costs.

9) For men, a BMI of 20 to 24.9 is considered normal. Above that, it is considered overweight, and below it is considered underweight. Create a new column called "bmi_male" that will receive the text "below normal" for BMI below 20, "normal" for BMI from 20 to 24.9, and "above normal" for BMI above 24.9.

In [21]:
conditions = [
    (df["sex"] == "male") & (df["bmi"] < 20),
    (df["sex"] == "male") & (df["bmi"] >= 20) & (df["bmi"] <= 24.9),
    (df["sex"] == "male") & (df["bmi"] > 24.9)
]

choices = ["below normal", "normal", "above normal"]

df["bmi_male"] = np.select(conditions, choices, default="not male")
